picard/ui/mainwindow.py

```diff
@@ -818,7 +818,12 @@ def save(self):
 
     def remove(self):
         """Tell the tagger to remove the selected objects."""
-        self.panel.remove(self.selected_objects)
+        if isinstance(self.selected_objects[0], ClusterList):
```

Comment:
This is the completely wroing place to add remove logic. All the remove logic exists in tagger,py#607.

Move this to that. Probably what is causing the spotty bugs.